### PR TITLE
E2E Tests: Migrate style variations tests to Playwright

### PR DESCRIPTION
**Pull request by ramonjd**
## What?
An attempt to chip in and migrate a playwright test. 

This might go horribly wrong and blow up.

## Why?
Part of the effort to move to Playwright. See: https://github.com/WordPress/gutenberg/issues/38851

## How?
Reading https://playwright.dev/docs 😄 

## Testing Instructions

You might have to restart wp-env after checking out this branch, then run:

```bash
npm run test-e2e:playwright -- site-editor/style-variations.spec.js
```

Screen Shot 2022-06-06 at 11 42 57 am

### CODE REVIEW

File: .wp-env.json [hunk 1]
```diff
@@ -1,14 +1,14 @@
 {
 	"core": "WordPress/WordPress",
 	"plugins": [ "." ],
-	"themes": [ "./test/emptytheme" ],
+	"themes": [ "./test/emptytheme", "./test/themes/style-variations" ],
```
Comment (ramonjd) [inline .wp-env.json]: I'm pretty sure this shouldn't be here. I added it so the tests pass for now until I can work out the correct entries for `tests.mappings`

File: .wp-env.json [hunk 2]
```diff
@@ -8,7 +8,7 @@
 				"wp-content/plugins/gutenberg": ".",
 				"wp-content/mu-plugins": "./packages/e2e-tests/mu-plugins",
 				"wp-content/plugins/gutenberg-test-plugins": "./packages/e2e-tests/plugins",
-				"wp-content/themes/gutenberg-test-themes": "./packages/e2e-tests/themes"
+				"wp-content/themes": "./test/gutenberg-test-themes"
```
Comment (ramonjd) [inline .wp-env.json]: We'll need to `encodeURIComponent` in `activateTheme()` to support theme names like gutenberg-test-themes/style-variations

File: packages/e2e-test-utils-playwright/src/editor/site-editor.ts [hunk 3]
```diff
@@ -23,6 +23,37 @@ export async function saveSiteEditorEntities( this: Editor ) {
...
+	);
+}
+
+/**
+ * Toggles the global styles sidebar (opens it if closed and closes it if open).
+ *
+ * @param {Editor} this
+ */
+export async function toggleGlobalStyles( this: Editor ) {
+	await this.page.click( 'role=button[name="Styles"i]' );
```
Comment (kevin940726) [inline packages/e2e-test-utils-playwright/src/editor/site-editor.ts]: Since this util is just one line and easily readable, could we just inline it?

File: packages/e2e-test-utils-playwright/src/editor/site-editor.ts [hunk 4]
```diff
@@ -23,6 +23,37 @@ export async function saveSiteEditorEntities( this: Editor ) {
...
+}
+
+/**
+ * Opens the previous global styles panel.
+ *
+ * @param {Editor} this
+ */
+export async function openPreviousGlobalStylesPanel( this: Editor ) {
+	await this.page.click(
+		'role=button[name="Navigate to the previous view"i]'
```
Comment (kevin940726) [inline packages/e2e-test-utils-playwright/src/editor/site-editor.ts]: Ditto to this util.

File: packages/e2e-test-utils-playwright/src/editor/site-editor.ts [hunk 5]
```diff
@@ -23,6 +23,37 @@ export async function saveSiteEditorEntities( this: Editor ) {
...
+}
+
+/**
+ * Opens a global styles panel.
+ *
+ * @param {Editor} this
+ * @param {string} panelName Name of the panel that is going to be opened.
+ */
+export async function openGlobalStylesPanel( this: Editor, panelName: string ) {
+	const selector = `xpath=//div[@aria-label="Settings"]//button[.//*[text()="${ panelName }"]]`;
```
Comment (kevin940726) [inline packages/e2e-test-utils-playwright/src/editor/site-editor.ts]: We usually want to use role selectors over other selectors. Furthermore, I think we can inline this as well if we convert it to use role selectors.

File: packages/e2e-test-utils-playwright/src/request-utils/themes.ts [hunk 6]
```diff
@@ -13,9 +13,9 @@ async function activateTheme(
 	let response = await this.request.get( THEMES_URL );
 	const html = await response.text();
 	const matchGroup = html.match(
-		new RegExp(
-			`action=activate&amp;stylesheet=${ themeSlug }&amp;_wpnonce=[a-z0-9]+`
-		)
+		`action=activate&amp;stylesheet=${ encodeURIComponent(
```
Comment (kevin940726) [inline packages/e2e-test-utils-playwright/src/request-utils/themes.ts:16]: Nice catch! 💯

File: test/e2e/specs/site-editor/style-variations.spec.js [hunk 7]
```diff
@@ -0,0 +1,299 @@
...
+		await admin.visitSiteEditor( {
+			postId: 'gutenberg-test-themes/style-variations//index',
+			postType: 'wp_template',
+		} );
+		await editor.toggleGlobalStyles();
+		await siteEditorStyleVariations.openOtherStyles();
+
+		const variations = await siteEditorStyleVariations.getAvailableStyleVariations();
+
+		expect( await variations.count() ).toEqual( 3 );
```
Comment (kevin940726) [inline test/e2e/specs/site-editor/style-variations.spec.js]: Let's use [Playwright-style assertions](https://playwright.dev/docs/test-assertions#locator-assertions-to-have-count):

```suggestion
		await expect( variations ).toHaveCount( 3 );
```

File: test/e2e/specs/site-editor/style-variations.spec.js [hunk 8]
```diff
@@ -0,0 +1,299 @@
...
+		expect( await variations.first().getAttribute( 'class' ) ).toContain(
+			'is-active'
+		);
+		expect(
+			await variations.nth( 1 ).getAttribute( 'class' )
+		).not.toContain( 'is-active' );
+		expect(
+			await variations.nth( 2 ).getAttribute( 'class' )
+		).not.toContain( 'is-active' );
```
Comment (kevin940726) [inline test/e2e/specs/site-editor/style-variations.spec.js]: I think it's a good opportunity for us to rethink the assertions here. Generally, we don't want to assert class names since they are often just implementation details. Instead, is there any accessible attribute we could get here to represent _active_? Like `aria-*` attribute or even `isVisible()` check.

If that's not possible for now, then I think we can safely say that the app is also not accessible to screen readers and we should fix that in a future PR. In that case, we can continue asserting class names here but we should add some comments above explaining why.

File: test/e2e/specs/site-editor/style-variations.spec.js [hunk 9]
```diff
@@ -0,0 +1,299 @@
...
+			postId: 'gutenberg-test-themes/style-variations//index',
+			postType: 'wp_template',
+		} );
+		await siteEditorStyleVariations.applyPinkVariation();
+		await editor.openPreviousGlobalStylesPanel();
+
+		await siteEditorStyleVariations.openColorsPanel();
+		expect(
+			await siteEditorStyleVariations.getBackgroundColorValue()
+		).toBe( 'rgb(202, 105, 211)' );
```
Comment (kevin940726) [inline test/e2e/specs/site-editor/style-variations.spec.js]: I wonder if we could instead use [`.toHaveCSS()`](https://playwright.dev/docs/test-assertions#locator-assertions-to-have-css) here?

File: test/e2e/specs/site-editor/style-variations.spec.js [hunk 10]
```diff
@@ -0,0 +1,299 @@
...
+	} ) => {
+		await admin.visitSiteEditor( {
+			postId: 'gutenberg-test-themes/style-variations//index',
+			postType: 'wp_template',
+		} );
+		await siteEditorStyleVariations.applyYellowVariation();
+		const frame = await page.frameLocator(
+			'css=.edit-site-visual-editor iframe'
+		);
+		const paragraph = await frame.locator(
```
Comment (kevin940726) [inline test/e2e/specs/site-editor/style-variations.spec.js]: Neither `.locator` nor `.frameLocator` returns a promise. It's a common mistake when first migrating to Playwright 😆 .

**Review by kevin940726 — COMMENTED** [review]
Thanks for starting this! Remember to checkout the [best practices](https://github.com/WordPress/gutenberg/blob/trunk/test/e2e/README.md#best-practices) guide and follow them wherever possible. I think this has a great start though!

File: test/e2e/specs/site-editor/style-variations.spec.js (hunk 9 above)
Comment (ramonjd) [reply test/e2e/specs/site-editor/style-variations.spec.js]: Ooh, nice! Thanks for the tip.

**Comment by ramonjd** [comment]
Thanks for helping with this one @kevin940726 !!  I'm learning a lot.

The first pass was just pulling everything across and getting things to work. 

I've added a bunch of aria-labels to navigation buttons to help tighten up the tests.

There are also still a few things left to do, such as refactoring the [class name assertions](https://github.com/WordPress/gutenberg/pull/41427#discussion_r889976831) as you recommend, and probably inlining some more methods. I'll ping for another 👀 after I finish that if you don't mind. 🙇

File: test/e2e/specs/site-editor/style-variations.spec.js (hunk 8 above)
Comment (ramonjd) [reply test/e2e/specs/site-editor/style-variations.spec.js]: Good advice, thanks. I'll look at this next.

Comment (ramonjd) [reply test/e2e/specs/site-editor/style-variations.spec.js]: I'm taking care of adding aria attributes over in https://github.com/WordPress/gutenberg/pull/41591

I probably would have done it here, but I found a bug as well where the active state cannot be accessed for some theme.json files.

So I think we can leave this until https://github.com/WordPress/gutenberg/pull/41591 is merged. I'll add a comment to that effect here.

**Comment by ramonjd** [comment]
🤞 I think this is in better shape @kevin940726, thanks for the pointers.

Once we land https://github.com/WordPress/gutenberg/pull/41591 I can refactor the classname assertions.

File: test/e2e/specs/site-editor/style-variations.spec.js [hunk 11]
```diff
@@ -0,0 +1,266 @@
...
+		this.pageUtils = pageUtils;
+		this.requestUtils = requestUtils;
+	}
+
+	async openOtherStyles() {
+		await this.page
+			.locator( 'role=button[name="Browse styles"i]' )
+			.waitFor( { state: 'attached' } ); // Wait for the element to appear after the navigation animation.
+
+		await this.page.click( 'role=button[name="Browse styles"i]' );
```
Comment (kevin940726) [inline test/e2e/specs/site-editor/style-variations.spec.js]: Playwright will auto-wait for the button to be clickable. I don't think we need to explicitly wait for it to be attached. This brings up down to this:

```suggestion
		await this.page.click( 'role=button[name="Browse styles"i]' );
```

Which I think we can just inline it without having to create an util here?

File: test/e2e/specs/site-editor/style-variations.spec.js [hunk 12]
```diff
@@ -0,0 +1,266 @@
...
+class SiteEditorStyleVariations {
+	constructor( { admin, editor, page, pageUtils, requestUtils } ) {
+		this.admin = admin;
+		this.editor = editor;
+		this.page = page;
+		this.pageUtils = pageUtils;
+		this.requestUtils = requestUtils;
+	}
+
+	async openOtherStyles() {
```
Comment (kevin940726) [inline test/e2e/specs/site-editor/style-variations.spec.js]: If we want to keep this util, then I think maybe `browseStyles()` would be a better name for this? It matches the name of the button being clicked.

File: test/e2e/specs/site-editor/style-variations.spec.js [hunk 13]
```diff
@@ -0,0 +1,266 @@
...
+	} ) => {
+		await admin.visitSiteEditor( {
+			postId: 'gutenberg-test-themes/style-variations//index',
+			postType: 'wp_template',
+		} );
+		await page.click( 'role=button[name="Styles"i]' );
+		await siteEditorStyleVariations.openOtherStyles();
+
+		const variations = page.locator(
+			'.edit-site-global-styles-variations_item'
```
Comment (kevin940726) [inline test/e2e/specs/site-editor/style-variations.spec.js:48]: Let's not depend on class names in our selectors. Other than fixing the `aria-label` and `aria-current` of the buttons, maybe we should also group them under an accessible section too.

image

We already have an `h5` and a `p` in the styles region, but they are not associated with any section or with each other. Maybe we could group them in a `<section>` or something else and add `aria-labelledby` and `aria-describedby` etc to provide clear accessible information?

It sounds like a bigger change than this PR though, so I agree that we can defer this to another PR and just an explanation comment above here for now.

File: test/e2e/specs/site-editor/style-variations.spec.js [hunk 14]
```diff
@@ -0,0 +1,266 @@
...
+		await siteEditorStyleVariations.applyPinkVariation();
+		await siteEditorStyleVariations.openPreviousGlobalStylesPanel();
+
+		await siteEditorStyleVariations.openColorsPanel();
+
+		await expect(
+			page.locator(
+				'css=[aria-label="Colors background styles"] .component-color-indicator'
+			)
+		).toHaveCSS( 'background', /rgb\(202, 105, 211\)/ );
```
Comment (kevin940726) [inline test/e2e/specs/site-editor/style-variations.spec.js:87]: Why do we have to use a regex here rather than just a string?

File: test/e2e/specs/site-editor/style-variations.spec.js [hunk 15]
```diff
@@ -0,0 +1,266 @@
...
+			postType: 'wp_template',
+		} );
+		await siteEditorStyleVariations.applyPinkVariation();
+		await siteEditorStyleVariations.openPreviousGlobalStylesPanel();
+
+		await siteEditorStyleVariations.openColorsPanel();
+
+		await expect(
+			page.locator(
+				'css=[aria-label="Colors background styles"] .component-color-indicator'
```
Comment (kevin940726) [inline test/e2e/specs/site-editor/style-variations.spec.js]: It might be worth it to add `data-testid` to the `component-color-indicator` so that we don't have to use class names here. Could be in another PR though.

File: test/e2e/specs/site-editor/style-variations.spec.js [hunk 16]
```diff
@@ -0,0 +1,266 @@
...
+	async openTextPanel() {
+		await this.openGlobalStylesPanel( 'Typography Text styles' );
+	}
+
+	async getFontSizeHint() {
+		const element = this.page.locator(
+			'css=.components-font-size-picker__header__hint'
+		);
+		return await element.evaluate( ( el ) => el.textContent );
+	}
```
Comment (kevin940726) [inline test/e2e/specs/site-editor/style-variations.spec.js]: I wonder if we could query the element by its accessible role and name and use Playwright-style assertions directly?

```js
await expect( elementLocator ).toHaveText( '...' );
```

If the element is accessible, we could probably even inline this directly in the tests.

File: test/e2e/specs/site-editor/style-variations.spec.js [hunk 17]
```diff
@@ -0,0 +1,266 @@
...
+
+	async getCustomFontSizeValue() {
+		const element = this.page.locator(
+			'css=.components-font-size-picker input[aria-label="Custom"]'
+		);
+		return await element.evaluate( ( el ) => el.value );
+	}
```
Comment (kevin940726) [inline test/e2e/specs/site-editor/style-variations.spec.js]: Same here:

```js
await expect( page.locator( 'role=spinbutton[name="Custom"i]' ) ).toHaveValue( '...' );
```

File: test/e2e/specs/site-editor/style-variations.spec.js [hunk 18]
```diff
@@ -0,0 +1,266 @@
...
+
+	async openPreviousGlobalStylesPanel() {
+		await this.page.click(
+			'role=button[name="Navigate to the previous view"i]'
+		);
+	}
```
Comment (kevin940726) [inline test/e2e/specs/site-editor/style-variations.spec.js]: Since that role selectors are already pretty readable, I'd suggest inlining this directly to avoid the readers having to jump between the utils and their implementations that often.

Same for other one/two-line utils in this file.

File: test/e2e/specs/site-editor/style-variations.spec.js [hunk 19]
```diff
@@ -0,0 +1,266 @@
...
+		const frame = page.frameLocator(
+			'css=.edit-site-visual-editor iframe'
+		);
+		const paragraph = frame.locator(
+			'xpath=//p[text()="My awesome paragraph"]'
+		);
+		const paragraphColor = await paragraph.evaluate( ( el ) => {
+			return window.getComputedStyle( el ).color;
+		} );
+		expect( paragraphColor ).toBe( 'rgb(25, 25, 17)' );
```
Comment (kevin940726) [inline test/e2e/specs/site-editor/style-variations.spec.js]: We can use the text selectors here to avoid weird XPath selectors. We can also leverage `toHaveCSS` to assert the color.

File: test/e2e/specs/site-editor/style-variations.spec.js [hunk 20]
```diff
@@ -0,0 +1,266 @@
...
+		const body = frame.locator( 'css=body' );
+		const backgroundColor = await body.evaluate( ( el ) => {
+			return window.getComputedStyle( el ).backgroundColor;
+		} );
+		expect( backgroundColor ).toBe( 'rgb(255, 239, 11)' );
```
Comment (kevin940726) [inline test/e2e/specs/site-editor/style-variations.spec.js]: Same here. Let's use `toHaveCSS`.

```suggestion
		await expect( body ).toHaveCSS( 'background-color', 'rgb(255, 239, 11)' );
```

**Review by kevin940726 — COMMENTED** [review]
Thanks for also fixing the aria issues!

File: test/e2e/specs/site-editor/style-variations.spec.js (hunk 11 above)
Comment (ramonjd) [reply test/e2e/specs/site-editor/style-variations.spec.js]: Yeah I originally tried that because I was experiencing a lot of timeouts, which I assumed were due to the animation.

`waiting for selector "role=button[name="Browse styles"i]"`

But the one liner seems to work locally for me 🤞 

Thanks!

File: test/e2e/specs/site-editor/style-variations.spec.js (hunk 16 above)
Comment (ramonjd) [reply test/e2e/specs/site-editor/style-variations.spec.js]: Nice. 

Yeah, these elements in the source code probably also need some work in relation to accessibility. 

In the interests of keeping this PR as small as possible, I'll keep the selectors as they are and make a note in the comments.

File: test/e2e/specs/site-editor/style-variations.spec.js (hunk 14 above)
Comment (ramonjd) [reply test/e2e/specs/site-editor/style-variations.spec.js:87]: The value of `background` contains other computed properties we don't care about in this test: `none repeat scroll 0% 0% / auto padding-box border-box`.

I should have used `background-color` in this case 👍

File: test/e2e/specs/site-editor/style-variations.spec.js (hunk 13 above)
Comment (ramonjd) [reply test/e2e/specs/site-editor/style-variations.spec.js:48]: Thanks for the suggestion. 

I think it's better left to another PR that can be reviewed on its own merits rather than in an E2E migration PR.

I'm happy to take care of it and circle back. 👍

File: test/e2e/specs/site-editor/style-variations.spec.js (hunk 11 above)
Comment (kevin940726) [reply test/e2e/specs/site-editor/style-variations.spec.js]: Do you remember why it was timing out before? Might want to keep an eye on that in case we accidentally ship a flaky test 😅.

File: test/e2e/specs/site-editor/style-variations.spec.js [hunk 21]
```diff
@@ -0,0 +1,236 @@
...
+
+		await expect(
+			page.locator(
+				'css=[aria-label="Colors background styles"] [data-testid="background-color-indicator"]'
+			)
+		).toHaveCSS( 'background', /rgb\(202, 105, 211\)/ );
+
+		await expect(
+			page.locator(
+				'css=[aria-label="Colors text styles"] [data-testid="text-color-indicator"]'
```
Comment (kevin940726) [inline test/e2e/specs/site-editor/style-variations.spec.js]: There's actually a [selector](https://playwright.dev/docs/selectors#id-data-testid-data-test-id-data-test-selectors) specifically for data-testid 😆 :

```suggestion
				'css=[aria-label="Colors text styles"] data-testid=text-color-indicator'
```

Also, any reason why we're using a css selector for `'css=[aria-label="Colors text styles"]` rather than a role selector?

Ditto to one above.

File: test/e2e/specs/site-editor/style-variations.spec.js [hunk 22]
```diff
@@ -0,0 +1,236 @@
...
+		siteEditorStyleVariations,
+	} ) => {
+		await admin.visitSiteEditor( {
+			postId: 'gutenberg-test-themes/style-variations//index',
+			postType: 'wp_template',
+		} );
+		await siteEditorStyleVariations.applyYellowVariation();
+		const frame = page.frameLocator(
+			'css=.edit-site-visual-editor iframe'
+		);
```
Comment (kevin940726) [inline test/e2e/specs/site-editor/style-variations.spec.js]: Maybe try `editor.canvas`?

File: test/e2e/specs/site-editor/style-variations.spec.js [hunk 23]
```diff
@@ -0,0 +1,236 @@
...
+		await this.page.click( `role=button[name="${ label }"i]` );
+	}
+
+	async applyPinkVariation() {
+		await this.applyVariation( 'pink' );
+	}
+
+	async applyYellowVariation() {
+		await this.applyVariation( 'yellow' );
+	}
```
Comment (kevin940726) [inline test/e2e/specs/site-editor/style-variations.spec.js]: Nit: How about we inline these as well?

File: test/e2e/specs/site-editor/style-variations.spec.js (hunk 14 above)
Comment (kevin940726) [reply test/e2e/specs/site-editor/style-variations.spec.js:87]: Oh, I see. Good call!

File: test/e2e/specs/site-editor/style-variations.spec.js (hunk 11 above)
Comment (ramonjd) [reply test/e2e/specs/site-editor/style-variations.spec.js]: > Do you remember why it was timing out before?

No. I was watching the reply in the browser and it I could see the element on the page, but playwright couldn't 🤣 

I only assumed it had something to do with the animation delay 🤷 

I just run the tests 10 x locally with the update and they pass. Let's see....

Comment (ramonjd) [reply test/e2e/specs/site-editor/style-variations.spec.js]: Oh, and now the 11th time:

```bash
    page.click: Timeout 10000ms exceeded.
    =========================== logs ===========================
    waiting for selector "role=button[name="Browse styles"i]"
    ============================================================

      221 | 	async applyVariation( label ) {
      222 | 		await this.page.click( 'role=button[name="Styles"i]' );
    > 223 | 		await this.page.click( 'role=button[name="Browse styles"i]' );
          | 		                ^
      224 | 		await this.page.click( `role=button[name="${ label }"i]` );
      225 | 	}
      226 |
``` 

😮‍💨

Comment (ramonjd) [reply test/e2e/specs/site-editor/style-variations.spec.js]: Ah, the Browse Styles button appears after a slight delay 😄

File: test/e2e/specs/site-editor/style-variations.spec.js (hunk 22 above)
Comment (ramonjd) [reply test/e2e/specs/site-editor/style-variations.spec.js]: `page.frame( 'editor-canvas' )` works too!

File: test/e2e/specs/site-editor/style-variations.spec.js (hunk 21 above)
Comment (ramonjd) [reply test/e2e/specs/site-editor/style-variations.spec.js]: Oh nice!

`'role=button[name="Colors background styles"i] >> data-testid=background-color-indicator'` seems to work.

File: test/e2e/specs/site-editor/style-variations.spec.js (hunk 11 above)
Comment (ramonjd) [reply test/e2e/specs/site-editor/style-variations.spec.js]: Ooooooooooooooooooh 🤦 🤦 🤦 🤦 🤦 🤦 🤦 🤦 🤦 

Screen Shot 2022-06-17 at 12 05 21 pm

**Comment by ramonjd** [comment]
I think this is nearly ready! Thanks for your patience @kevin940726 

Dismissing the styles welcome guide appears to have done the trick!

File: test/e2e/specs/site-editor/style-variations.spec.js [hunk 24]
```diff
@@ -0,0 +1,226 @@
...
+		await siteEditorStyleVariations.browseStyles();
+		await page.click( 'role=button[name="yellow"i]' );
+		await page.click(
+			'role=button[name="Navigate to the previous view"i]'
+		);
+		await page.click( 'role=button[name="Colors styles"i]' );
+
+		await expect(
+			page.locator(
+				'css=[aria-label="Colors background styles"] [data-testid="background-color-indicator"]'
```
Comment (kevin940726) [inline test/e2e/specs/site-editor/style-variations.spec.js]: Same here and the one below 😅

File: test/e2e/specs/site-editor/style-variations.spec.js [hunk 25]
```diff
@@ -0,0 +1,226 @@
...
+			)
+		).toHaveCSS( 'background', /rgb\(255, 239, 11\)/ );
+
+		await expect(
+			page.locator(
+				'css=[aria-label="Colors text styles"] [data-testid="text-color-indicator"]'
```
Comment (kevin940726) [inline test/e2e/specs/site-editor/style-variations.spec.js]: Ditto.

File: test/e2e/specs/site-editor/style-variations.spec.js [hunk 26]
```diff
@@ -0,0 +1,226 @@
...
+
+		// TODO: to avoid use classnames to locate these elements,
+		//  we could provide accessible attributes to the source code in packages/components/src/font-size-picker/index.js.
+		await expect(
+			page.locator( 'css=.components-font-size-picker__header__hint' )
+		).toHaveText( '(Custom)' );
+
+		await expect(
+			page.locator(
+				'css=.components-font-size-picker input[aria-label="Custom"]'
```
Comment (kevin940726) [inline test/e2e/specs/site-editor/style-variations.spec.js]: Don't we already have accessible role and name for this element?

**Review by kevin940726 — APPROVED** [review]
Awesome, thanks! I think you missed some selectors (😅) but overall it's pretty good! Feel free to merge it after fixing those places 👍

File: test/e2e/specs/site-editor/style-variations.spec.js (hunk 24 above)
Comment (ramonjd) [reply test/e2e/specs/site-editor/style-variations.spec.js]: Whoops! Thanks

File: test/e2e/specs/site-editor/style-variations.spec.js (hunk 26 above)
Comment (ramonjd) [reply test/e2e/specs/site-editor/style-variations.spec.js]: Sorry, I forgot about 'spinbutton'

All I see are green ✅  :trollface:
